### PR TITLE
Enabled reply-to and formatted email addresses (Name <email>) in poste.

### PR DESCRIPTION
**Pull request by aoby**
Added a unit test for poste. Fixed up some typos and updated some text
in the emails.
Also changed scheduled mails to go out at 7:30am PST (14:30 UTC) instead of 9:30am.

Note 'Reply-To' is consumed [here](https://github.com/code-dot-org/code-dot-org/blob/4542a06f0904d1b743c4c519917ba720517278c6/bin/deliver_poste_messages#L138-L138).

### CODE REVIEW

File: dashboard/app/mailers/pd/workshop_mailer.rb [hunk 1]
```diff
@@ -96,9 +96,9 @@ def exit_survey(workshop, teacher, enrollment)
     @survey_url = CDO.code_org_url "/pd-workshop-survey/#{enrollment.code}"
 
     mail content_type: 'text/html',
+      from: email_address('Hadi Partovi', 'hadi_partovi@code.org'),
```
Comment (mehalshah) [inline dashboard/app/mailers/pd/workshop_mailer.rb]: Might as well make a Hadi method for consistency

File: dashboard/app/views/pd/workshop_mailer/exit_survey.html.haml [hunk 2]
```diff
@@ -22,7 +22,7 @@
     the opportunity to learn computer science.
 
 %p
-  = link_to('Please take a minute to complete this short survey', @survey_url)
+  = link_to('Please take a minute to complete this short survey', @survey_url) + '.'
```
Comment (mehalshah) [inline dashboard/app/views/pd/workshop_mailer/exit_survey.html.haml:25]: Nit - any reason the period isn't clickable?

File: lib/cdo/poste.rb [hunk 3]
```diff
@@ -203,21 +203,26 @@ def self.send_message(message_name, recipient, params)
...
-      raise ArgumentError, "Unsupported sender: #{sender}" unless ALLOWED_SENDERS.include?(sender)
+      sender_email = mail.from.first
+      raise ArgumentError, "Unsupported sender: #{sender_email}" unless ALLOWED_SENDERS.include?(sender_email)
 
+      sender = mail[:from].formatted.first
+      reply_to = mail[:reply_to].formatted.first
       subject = mail.subject.to_s
       body = mail.body.to_s
+      to_address = mail[:to].addresses.first
+      to_name = mail[:to].display_names.first
```
Comment (mehalshah) [inline lib/cdo/poste.rb:220]: Handle addresses / display_names being nil or empty

**Comment by mehalshah** [comment]
couple comments, rest lgtm

File: dashboard/app/views/pd/workshop_mailer/exit_survey.html.haml (hunk 2 above)
Comment (tanyaparker) [inline dashboard/app/views/pd/workshop_mailer/exit_survey.html.haml:25]: I tend to prefer that end punctuation isn't included in the hyperlink.

Comment (aoby) [inline dashboard/app/views/pd/workshop_mailer/exit_survey.html.haml:25]: At this point, that's how all the other links are so I'm keeping with the pattern. I initially picked it up from the [spec](https://docs.google.com/document/d/1blKBDZ167AoB01C4GcQ75yi0Jc4G2AW42imsenirvMk) which shows the period outside the link. I also personally prefer it that way (I think of the link as part of the sentence content, and the period as part of the container), but I don't feel too strongly about it.

File: dashboard/app/mailers/pd/workshop_mailer.rb (hunk 1 above)
Comment (aoby) [inline dashboard/app/mailers/pd/workshop_mailer.rb]: will do

File: lib/cdo/poste.rb (hunk 3 above)
Comment (aoby) [inline lib/cdo/poste.rb:220]: good suggestion. I'll add that to the test as well.

**Comment by aoby** [comment]
Updated. 
I also realized that the email template was setting the from address to `"Code.org <#{from}>"`, so I had to update that as well or else I'd get something like `"Code.org <Code.org <teacher@code.org>>"` Please take another quick look.

**Comment by mehalshah** [comment]
lgtm

**Pull request merged**